src/bindings/java/compile-native-files.cmake:71: Could not compile java wrappers: error: Source option 6 is no longer supported. Use 7 or later.

Hello.

Java binding is not compiling with `java-17-openjdk 17.0.2.0.8`:

```
make[2]: Entering directory '/builddir/build/BUILD/libSBML-5.19.0-Source/build'
[  2%] Building: libsbml.jar
cd /builddir/build/BUILD/libSBML-5.19.0-Source/build/src/bindings/java && /usr/bin/cmake -E make_directory /builddir/build/BUILD/libSBML-5.19.0-Source/build/src/bindings/java/test
cd /builddir/build/BUILD/libSBML-5.19.0-Source/build/src/bindings/java && /usr/bin/cmake -DJava_JAVAC_EXECUTABLE="/usr/bin/javac" -DJava_JAR_EXECUTABLE="/usr/bin/jar" -DBIN_DIRECTORY="/builddir/build/BUILD/libSBML-5.19.0-Source/build/src/bindings/java" -DSRC_DIRECTORY="/builddir/build/BUILD/libSBML-5.19.0-Source/src/bindings/java" -DFILE_SEP=":" -DPATH_SEP=/ -DCOMPATIBILIY=1.6 -P /builddir/build/BUILD/libSBML-5.19.0-Source/src/bindings/java/compile-native-files.cmake
Creating: libsbml.jar
CMake Error at /builddir/build/BUILD/libSBML-5.19.0-Source/src/bindings/java/compile-native-files.cmake:71 (message):
  
  Could not compile java wrappers:
  
  warning: [options] bootstrap class path not set in conjunction with -source
  6
  error: Source option 6 is no longer supported.  Use 7 or later.
  error: Target option 6 is no longer supported.  Use 7 or later.
  
make[2]: *** [src/bindings/java/CMakeFiles/binding_java_jar.dir/build.make:77: src/bindings/java/libsbmlj.jar] Error 1
make[2]: Leaving directory '/builddir/build/BUILD/libSBML-5.19.0-Source/build'
make[1]: *** [CMakeFiles/Makefile2:2406: src/bindings/java/CMakeFiles/binding_java_jar.dir/all] Error 2
```

--- a/examples/java/CMakeLists.txt
+++ b/examples/java/CMakeLists.txt
@@ -35,6 +35,9 @@ file(TO_NATIVE_PATH ${CMAKE_CURRENT_BINARY_DIR} CURRENT_BIN)
 file(TO_NATIVE_PATH ${CMAKE_CURRENT_BINARY_DIR}/../../src/bindings/java/libsbmlj.jar CURRENT_JAR )
 file(TO_NATIVE_PATH $<TARGET_FILE_DIR:binding_java_lib> LIBRARY_PATH)
 
+if (JAVA_COMPATIBILITY)
+SET(COMPAT_ARGS "-source ${JAVA_COMPATIBILITY} -target ${JAVA_COMPATIBILITY}")
+endif()
 
 foreach(file 
     addCustomValidator
@@ -76,8 +79,7 @@ add_custom_command(
     COMMAND "${Java_JAVAC_EXECUTABLE}"
     ARGS -cp ".${FILE_SEP}\"${CURRENT_JAR}\"${FILE_SEP}${CMAKE_CURRENT_SOURCE_DIR}"
         ${CURRENT_FILE}
-        -source "${JAVA_COMPATIBILITY}"
-        -target "${JAVA_COMPATIBILITY}"
+        ${COMPAT_ARGS}
         -d ${CMAKE_CURRENT_BINARY_DIR}
     MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/../../src/bindings/java/local.i"
     WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"

--- a/examples/java/arrays/CMakeLists.txt
+++ b/examples/java/arrays/CMakeLists.txt
@@ -6,6 +6,10 @@
 
 find_package(Java COMPONENTS Development REQUIRED)
 
+if (JAVA_COMPATIBILITY)
+SET(COMPAT_ARGS "-source ${JAVA_COMPATIBILITY} -target ${JAVA_COMPATIBILITY}")
+endif()
+
 foreach(file 
 		createArrays1
 		createArrays2
@@ -20,8 +24,7 @@ foreach(file
 		COMMAND "${Java_JAVAC_EXECUTABLE}"
 		ARGS -cp ".${FILE_SEP}\"${CURRENT_JAR}\"${FILE_SEP}${CMAKE_CURRENT_SOURCE_DIR}"
 			 ${CURRENT_FILE}
-			 -source "${JAVA_COMPATIBILITY}"
-			 -target "${JAVA_COMPATIBILITY}"
+			 ${COMPAT_ARGS}
 			 -d ${CMAKE_CURRENT_BINARY_DIR}
 		MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/../../../src/bindings/java/local.i"
 		WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"

--- a/examples/java/comp/CMakeLists.txt
+++ b/examples/java/comp/CMakeLists.txt
@@ -31,6 +31,10 @@
 
 find_package(Java COMPONENTS Development REQUIRED)
 
+if (JAVA_COMPATIBILITY)
+SET(COMPAT_ARGS "-source ${JAVA_COMPATIBILITY} -target ${JAVA_COMPATIBILITY}")
+endif()
+
 foreach(file 
 		flattenModel
 		customResolver
@@ -45,8 +49,7 @@ foreach(file
 		COMMAND "${Java_JAVAC_EXECUTABLE}"
 		ARGS -cp ".${FILE_SEP}\"${CURRENT_JAR}\"${FILE_SEP}${CMAKE_CURRENT_SOURCE_DIR}"
 			 ${CURRENT_FILE}
-			 -source "${JAVA_COMPATIBILITY}"
-			 -target "${JAVA_COMPATIBILITY}"
+			 ${COMPAT_ARGS}
 			 -d ${CMAKE_CURRENT_BINARY_DIR}
 		MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/../../../src/bindings/java/local.i"
 		WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"

--- a/examples/java/distrib/CMakeLists.txt
+++ b/examples/java/distrib/CMakeLists.txt
@@ -6,6 +6,10 @@
 
 find_package(Java COMPONENTS Development REQUIRED)
 
+if (JAVA_COMPATIBILITY)
+SET(COMPAT_ARGS "-source ${JAVA_COMPATIBILITY} -target ${JAVA_COMPATIBILITY}")
+endif()
+
 foreach(file 
 	
 	createNormalExample
@@ -22,8 +26,7 @@ foreach(file
 		COMMAND "${Java_JAVAC_EXECUTABLE}"
 		ARGS -cp ".${FILE_SEP}\"${CURRENT_JAR}\"${FILE_SEP}${CMAKE_CURRENT_SOURCE_DIR}"
 			 ${CURRENT_FILE}
-			 -source "${JAVA_COMPATIBILITY}"
-			 -target "${JAVA_COMPATIBILITY}"
+			 ${COMPAT_ARGS}
 			 -d ${CMAKE_CURRENT_BINARY_DIR}
 		MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/../../../src/bindings/java/local.i"
 		WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"

--- a/examples/java/fbc/CMakeLists.txt
+++ b/examples/java/fbc/CMakeLists.txt
@@ -31,6 +31,10 @@
 
 find_package(Java COMPONENTS Development REQUIRED)
 
+if (JAVA_COMPATIBILITY)
+SET(COMPAT_ARGS "-source ${JAVA_COMPATIBILITY} -target ${JAVA_COMPATIBILITY}")
+endif()
+
 foreach(file 
 		convertCobraToFbc
 		convertFbcToCobra
@@ -46,8 +50,7 @@ foreach(file
 		COMMAND "${Java_JAVAC_EXECUTABLE}"
 		ARGS -cp ".${FILE_SEP}\"${CURRENT_JAR}\"${FILE_SEP}${CMAKE_CURRENT_SOURCE_DIR}"
 			 ${CURRENT_FILE}
-			 -source "${JAVA_COMPATIBILITY}"
-			 -target "${JAVA_COMPATIBILITY}"
+			 ${COMPAT_ARGS}
 			 -d ${CMAKE_CURRENT_BINARY_DIR}
 		MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/../../../src/bindings/java/local.i"
 		WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"

--- a/examples/java/groups/CMakeLists.txt
+++ b/examples/java/groups/CMakeLists.txt
@@ -6,6 +6,10 @@
 
 find_package(Java COMPONENTS Development REQUIRED)
 
+if (JAVA_COMPATIBILITY)
+SET(COMPAT_ARGS "-source ${JAVA_COMPATIBILITY} -target ${JAVA_COMPATIBILITY}")
+endif()
+
 foreach(file 
 		groups_example1
 		)
@@ -19,8 +23,7 @@ foreach(file
 		COMMAND "${Java_JAVAC_EXECUTABLE}"
 		ARGS -cp ".${FILE_SEP}\"${CURRENT_JAR}\"${FILE_SEP}${CMAKE_CURRENT_SOURCE_DIR}"
 			 ${CURRENT_FILE}
-			 -source "${JAVA_COMPATIBILITY}"
-			 -target "${JAVA_COMPATIBILITY}"
+			 ${COMPAT_ARGS}
 			 -d ${CMAKE_CURRENT_BINARY_DIR}
 		MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/../../../src/bindings/java/local.i"
 		WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"

--- a/examples/java/layout/CMakeLists.txt
+++ b/examples/java/layout/CMakeLists.txt
@@ -31,6 +31,10 @@
 
 find_package(Java COMPONENTS Development REQUIRED)
 
+if (JAVA_COMPATIBILITY)
+SET(COMPAT_ARGS "-source ${JAVA_COMPATIBILITY} -target ${JAVA_COMPATIBILITY}")
+endif()
+
 foreach(file 
 		layout_example1
 		layout_example1_L3
@@ -46,8 +50,7 @@ foreach(file
 		COMMAND "${Java_JAVAC_EXECUTABLE}"
 		ARGS -cp ".${FILE_SEP}\"${CURRENT_JAR}\"${FILE_SEP}${CMAKE_CURRENT_SOURCE_DIR}"
 			 ${CURRENT_FILE}
-			 -source "${JAVA_COMPATIBILITY}"
-			 -target "${JAVA_COMPATIBILITY}"
+			 ${COMPAT_ARGS}
 			 -d ${CMAKE_CURRENT_BINARY_DIR}
 		MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/../../../src/bindings/java/local.i"
 		WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"

--- a/examples/java/multi/CMakeLists.txt
+++ b/examples/java/multi/CMakeLists.txt
@@ -31,6 +31,10 @@
 
 find_package(Java COMPONENTS Development REQUIRED)
 
+if (JAVA_COMPATIBILITY)
+SET(COMPAT_ARGS "-source ${JAVA_COMPATIBILITY} -target ${JAVA_COMPATIBILITY}")
+endif()
+
 foreach(file 
 		multi_example1
 		multi_example2
@@ -46,8 +50,7 @@ foreach(file
 		COMMAND "${Java_JAVAC_EXECUTABLE}"
 		ARGS -cp ".${FILE_SEP}\"${CURRENT_JAR}\"${FILE_SEP}${CMAKE_CURRENT_SOURCE_DIR}"
 			 ${CURRENT_FILE}
-			 -source "${JAVA_COMPATIBILITY}"
-			 -target "${JAVA_COMPATIBILITY}"
+			 ${COMPAT_ARGS}
 			 -d ${CMAKE_CURRENT_BINARY_DIR}
 		MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/../../../src/bindings/java/local.i"
 		WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"

--- a/examples/java/qual/CMakeLists.txt
+++ b/examples/java/qual/CMakeLists.txt
@@ -31,6 +31,10 @@
 
 find_package(Java COMPONENTS Development REQUIRED)
 
+if (JAVA_COMPATIBILITY)
+SET(COMPAT_ARGS "-source ${JAVA_COMPATIBILITY} -target ${JAVA_COMPATIBILITY}")
+endif()
+
 foreach(file 
 		qual_example1
 		)
@@ -44,8 +48,7 @@ foreach(file
 		COMMAND "${Java_JAVAC_EXECUTABLE}"
 		ARGS -cp ".${FILE_SEP}\"${CURRENT_JAR}\"${FILE_SEP}${CMAKE_CURRENT_SOURCE_DIR}"
 			 ${CURRENT_FILE}
-			 -source "${JAVA_COMPATIBILITY}"
-			 -target "${JAVA_COMPATIBILITY}"
+			 ${COMPAT_ARGS}
 			 -d ${CMAKE_CURRENT_BINARY_DIR}
 		MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/../../../src/bindings/java/local.i"
 		WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"

--- a/examples/java/render/CMakeLists.txt
+++ b/examples/java/render/CMakeLists.txt
@@ -6,6 +6,10 @@
 
 find_package(Java COMPONENTS Development REQUIRED)
 
+if (JAVA_COMPATIBILITY)
+SET(COMPAT_ARGS "-source ${JAVA_COMPATIBILITY} -target ${JAVA_COMPATIBILITY}")
+endif()
+
 foreach(file 
 		addRenderInformation
 		printRenderInformation
@@ -21,8 +25,7 @@ foreach(file
 		COMMAND "${Java_JAVAC_EXECUTABLE}"
 		ARGS -cp ".${FILE_SEP}\"${CURRENT_JAR}\"${FILE_SEP}${CMAKE_CURRENT_SOURCE_DIR}"
 			 ${CURRENT_FILE}
-			 -source "${JAVA_COMPATIBILITY}"
-			 -target "${JAVA_COMPATIBILITY}"
+			 ${COMPAT_ARGS}
 			 -d ${CMAKE_CURRENT_BINARY_DIR}
 		MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/../../../src/bindings/java/local.i"
 		WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"

--- a/examples/java/spatial/CMakeLists.txt
+++ b/examples/java/spatial/CMakeLists.txt
@@ -31,6 +31,10 @@
 
 find_package(Java COMPONENTS Development REQUIRED)
 
+if (JAVA_COMPATIBILITY)
+SET(COMPAT_ARGS "-source ${JAVA_COMPATIBILITY} -target ${JAVA_COMPATIBILITY}")
+endif()
+
 foreach(file 
 		spatial_example1
 		)
@@ -44,8 +48,7 @@ foreach(file
 		COMMAND "${Java_JAVAC_EXECUTABLE}"
 		ARGS -cp ".${FILE_SEP}\"${CURRENT_JAR}\"${FILE_SEP}${CMAKE_CURRENT_SOURCE_DIR}"
 			 ${CURRENT_FILE}
-			 -source "${JAVA_COMPATIBILITY}"
-			 -target "${JAVA_COMPATIBILITY}"
+			 ${COMPAT_ARGS}
 			 -d ${CMAKE_CURRENT_BINARY_DIR}
 		MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/../../../src/bindings/java/local.i"
 		WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"

--- a/src/bindings/java/CMakeLists.txt
+++ b/src/bindings/java/CMakeLists.txt
@@ -74,8 +74,8 @@ if (NOT EXISTS "${PYTHON_EXECUTABLE}")
 endif()
 
 
-set(JAVA_COMPATIBILITY "1.6" CACHE STRING
-  "Specify the source and target compatibility for the libsbml Java bindings. Leave empty to remove the parameter altogether.")
+set(JAVA_COMPATIBILITY "" CACHE STRING
+  "Specify the source and target compatibility for the libsbml Java bindings (for example 1.7). Leave empty to remove the parameter altogether.")
 
 
 ####################################################################

